reconstructSPI/iterative_refinement/expectation_maximization.py

```diff
@@ -476,14 +525,36 @@ def generate_slices(map_3d_f, xy_plane, rots):
         """
         n_rotations = len(rots)
         n_pix = len(map_3d_f)
-        slices = np.empty((n_rotations, n_pix, n_pix))
+        slices = np.empty((n_rotations, n_pix, n_pix), dtype=np.complex_)
         overwrite_empty_with_zero = 0
         slices[:, :, 0] = overwrite_empty_with_zero
-        xyz_rotated = np.empty((n_rotations, 3, n_pix**2))
+        xyz_rotated = np.empty((n_rotations, 3, 3 * n_pix**2))
+        offset = np.array(
+            [
+                [
+                    0,
+                ],
+                [
+                    0,
+                ],
+                [
+                    z_offset,
+                ],
+            ]
+        )
+        xy_plane = np.concatenate(
```

Comment:
I follow what this does: duplicated x and y coords, but has z offset to -0.05, 0, 0.05. 

But why do you need it? Where do you use these new offset points?